File: Android/app/src/main/java/ru/inwords/inwords/domain/interactor/translation/TranslationWordsRepositoryInteractor.java
package ru.inwords.inwords.domain.interactor.translation;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import ru.inwords.inwords.data.dto.WordTranslation;

public interface TranslationWordsRepositoryInteractor {
    Observable<WordTranslation> getTranslation(String word);

    Observable<WordTranslation> getByOne();

    Observable<List<WordTranslation>> getList();

    Completable add(WordTranslation wordTranslation);

    Completable addAll(List<WordTranslation> wordTranslations);

    Completable markRemoved(WordTranslation wordTranslation);

    Completable markRemovedAll(List<WordTranslation> wordTranslations);

    void clearCache();
}

File: Android/app/src/main/java/ru/inwords/inwords/domain/interactor/translation/TranslationWordsRepositoryInteractor.kt
package ru.inwords.inwords.domain.interactor.translation

import io.reactivex.Completable
import io.reactivex.Observable
import ru.inwords.inwords.data.dto.WordTranslation

interface TranslationWordsRepositoryInteractor {
    fun getByOne(): Observable<WordTranslation>

    fun getList(): Observable<List<WordTranslation>>

    fun add(wordTranslation: WordTranslation): Completable

    fun addAll(wordTranslations: List<WordTranslation>): Completable

    fun markRemoved(wordTranslation: WordTranslation): Completable

    fun markRemovedAll(wordTranslations: List<WordTranslation>): Completable

    fun clearCache()
}
